The original license of the source code:

# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at http://mozilla.org/MPL/2.0/.
----
The bintools package on Gentoo doesn't produce a
/usr/bin/i686-pc-linux-gnu-readelf (target) only
/usr/bin/x86_64-pc-linux-gnu-readelf (host).
----
--- a/python/mozbuild/mozbuild/action/check_binary.py.orig	2020-12-24 08:23:20.943860468 -0800
+++ b/python/mozbuild/mozbuild/action/check_binary.py	2020-12-24 08:27:02.712494937 -0800
@@ -36,7 +36,7 @@ HOST = {
 TARGET = {
     "MOZ_LIBSTDCXX_VERSION": buildconfig.substs.get("MOZ_LIBSTDCXX_TARGET_VERSION"),
     "platform": buildconfig.substs["OS_TARGET"],
-    "readelf": "{}readelf".format(buildconfig.substs.get("TOOLCHAIN_PREFIX", "")),
+    "readelf": "readelf",
 }
 
 ADDR_RE = re.compile(r"[0-9a-f]{8,16}")
